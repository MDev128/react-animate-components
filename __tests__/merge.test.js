import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { shallow } from "enzyme";

import Merge from "../src/containers/merge";

import { left, bounce } from "../index";

const renderh1 = (
  <Merge one={{ name: left }} two={{ name: bounce }} as="h1">
    Hello
  </Merge>
);

const defaultRender = (
  <Merge one={{ name: left }} two={{ name: bounce }} className="main">
    <p>Hello</p>
  </Merge>
);

it("animates the children and render the h1 element", () => {
  const tree = renderer.create(renderh1).toJSON;

  expect(tree).toMatchSnapshot();
});

it("animates the children and render the div (default) element", () => {
  const tree = renderer.create(defaultRender).toJSON;

  expect(tree).toMatchSnapshot();
});

it("matches the keyframes created by styled-components and default props", () => {
  const tree = renderer.create(defaultRender).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

it("calls componentDidMount lifecycle method", () => {
  const wrapper = shallow(defaultRender);
  wrapper.instance().componentDidMount();
  // Also calls store() method when component mounts.
});

it("updates the styles when the component mounts", () => {
  const wrapper = shallow(renderh1);
  wrapper.instance().componentDidMount();

  // Implicitly calls the returnAnimation()
  expect(wrapper.state("styles")).toEqual({
    animation: "fRXJaB 1s ease, bGJZDZ 1s ease",
    backfaceVisibility: "visible"
  });
});
