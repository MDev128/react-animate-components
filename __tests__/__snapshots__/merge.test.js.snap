// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Merge Component animates the children and render the div (default) element 1`] = `[Function]`;

exports[`Merge Component animates the children and render the h1 element 1`] = `[Function]`;

exports[`Merge Component matches the keyframes created by styled-components and default props 1`] = `


<div
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className="main"
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "fRXJaB 1s ease, bGJZDZ 1s ease",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Merge Component renders the Component passed through the component prop 1`] = `


<div
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className={undefined}
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "fRXJaB 1s ease, bGJZDZ 1s ease",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  <div>
    <p>
      Hello Clay Jansen!
    </p>
  </div>
</div>
`;
